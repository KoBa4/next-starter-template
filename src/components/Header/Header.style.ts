import styled from 'styled-components'

export const HeaderBlock = styled.header`
  margin-inline: auto;
  padding: 0;
`
