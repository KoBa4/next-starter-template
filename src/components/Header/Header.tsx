import { NextPage } from 'next'
import { HeaderBlock } from './Header.style'

const Header: NextPage = () => (
  <HeaderBlock data-testid="Header">
    <h1>Header</h1>
  </HeaderBlock>
)

export default Header
