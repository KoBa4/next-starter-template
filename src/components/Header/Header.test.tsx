import { render, screen } from '@testing-library/react'
import Header from './Header'

describe('Header component', () => {
  it('renders correctly', () => {
    render(<Header />)
    const heading = screen.getByRole('heading', {
      name: /Header/i,
    })
    expect(heading).toBeInTheDocument()
  })

  it('Header snapshot', () => {
    const { container } = render(<Header />)
    expect(container).toMatchSnapshot()
  })
})
