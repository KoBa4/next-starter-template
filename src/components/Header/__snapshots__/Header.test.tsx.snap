// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Header snapshot 1`] = `
<div>
  <header
    class="sc-bdvvtL ekkTz"
    data-testid="Header"
  >
    <h1>
      Header
    </h1>
  </header>
</div>
`;
